File: Python/7py 1/dative.py
"""
|--------------------------------------------------------------------------
| Hungarian Vowel Harmony (easy)
|--------------------------------------------------------------------------
|
"""

def dative(word):
   nek = ["e", "é", "i", "í", "ö", "ő", "ü", "ű"]
   nak = ["a", "á", "o", "ó", "u", "ú"]
   x = [i for i in word if i in nek+nak][-1]
   return f"{word}nek" if x in nek else f"{word}nak"


def dative(word):
   for x in reversed(word):
      if x in "aáoóuú": return word + "nak"
      if x in "eéiíöőüű": return word + "nek"


from re import sub
def dative(word):
   return word + ("nak" if sub("[^eéiíöőüűaáoóuú]", "", word)[-1] in u"aáoóuú" else "nek" )


print(dative("ablak"), "ablaknak")
print(dative("szék"), "széknek")
print(dative("otthon"), "otthonnak")
print(dative("tükör"), "tükörnek")

File: PHP/7php 1/dative.php
<?php
/*
|--------------------------------------------------------------------------
| Hungarian Vowel Harmony (easy)
|--------------------------------------------------------------------------
|
*/

function dative($word) {
   for ($i = strlen($word) - 1; $i >= 0; $i--) {
      if (strpos("aáoóuú", $word[$i]) !== false) return $word . "nak";
      if (strpos("eéiíöőüű", $word[$i]) !== false) return $word . "nek";
   }
}


function dative(string $w): string {
   return (preg_match('[e|é|i|í|ö|ő|ü|ű]', $w)> preg_match('[a|á|o|ó|u|ú]', $w))? $w."nek": $w."nak";
}


function dative(string $w): string {
   str_replace(['a', 'á', 'o', 'ó','u', 'ú'],'',$w,$count);
   return $w.($count > 0 ? 'nak' : 'nek');
}


print_r(dative("ablak") . " ablaknak
");
print_r(dative("szék") . " széknek
");
print_r(dative("otthon") . " otthonnak
");
print_r(dative("tükör") . " tükörnek
");